hw3/exercise3/src/components/Search.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+const url = "https://thronesapi.com/api/v2/Characters";
+function Search() {
+  const [charList, setCharList] = useState([]);
+  const [searchedChar, setSearchedChar] = useState("");
+
+  useEffect(() => {
+    axios.get(url).then((res) => setCharList(res.data));
+  }, []);
+
+  return (
+    <div class="container">
+      <h2>
+        <label for="searchInput" className="input-search-label">
+          Search Character:
+        </label>
+        <input
+          type="text"
+          id="searchInput"
+          className="input-search-char"
+          placeholder="Search Character here.."
+          onChange={(e) => setSearchedChar(e.target.value)}
+        />
+      </h2>
+
+      <section id="container-characters" class="mx-auto">
+        {charList
+          .filter((item) => {
+            if (searchedChar === "") {
+              return item;
+            } else if (
+              item.fullName.toUpperCase().includes(searchedChar.toUpperCase())
+            ) {
+              return item;
+            }
+          })
+          .map((item) => {
+            let altText = `${item.fullName} of ${item.family}`;
+            return (
+              <div class="card">
+                <div className="card-body">
+                  <img src={item.imageUrl} alt={altText} class="card-img" />
+                  <span className="card-title">{item.fullName}</span>
+                  <p className="card-text">{item.family}</p>
+                </div>
+              </div>
+            );
+          })}
```

Comment:
![shruti4](https://github.com/shrutidaundkar/frontend-homework/assets/38127011/9956708b-a4d3-4d1e-814f-1d69ebd68a7b)

Missing keys!!

JSX elements directly inside a map() call always need keys!

Keys tell React which array item each component corresponds to, so that it can match them up later. This becomes important if your array items can move (e.g. due to sorting), get inserted, or get deleted. A well-chosen key helps React infer what exactly has happened, and make the correct updates to the DOM tree.